doc/user-guide/duckarrays.rst

```diff
@@ -1,30 +1,170 @@
 .. currentmodule:: xarray
 
+.. _userguide.duckarrays:
+
 Working with numpy-like arrays
 ==============================
 
+NumPy-like arrays (often known as :term:`duck array`\s) are drop-in replacements for the :py:class:`numpy.ndarray`
+class but with different features, such as propagating physical units or a different layout in memory.
+Xarray can often wrap these array types, allowing you to use labelled dimensions and indexes whilst benefiting from the
+additional features of these array libraries.
+
+Some numpy-like array types that xarray already has some support for:
+
+* `Cupy <https://cupy.dev/>`_ - GPU support,
+* `Sparse <https://sparse.pydata.org/en/stable/>`_ - for performant arrays with many zero elements,
+* `Pint <https://pint.readthedocs.io/en/latest/>`_ - for tracking the physical units of your data.
```

Comment:
Let's also link to cupy-xarray and pint-xarray.

Reply:
Let's also add dask (and cubed)? to the list!

Reply:
I was actually going to do a follow-up PR adding internals documentation on wrapping other chunked arrays, but I'll mention dask and cubed here too.